include/dt-bindings/pinctrl/mchp-xec-pinctrl.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_MCHP_XEC_PINCTRL_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_PINCTRL_MCHP_XEC_PINCTRL_H_
+
+#include <dt-bindings/pinctrl/mchp-xec-pinctrl-common.h>
+
+/* Adapted from Linux: include/dt-bindings/pinctrl/stm32-pinfunc.h */
+
+/**
+ * @brief Pin modes
+ */
+
+/*
+ * Microchip XEC GPIO pins implement up to 8 functions.
+ * Function 0 is GPIO mode.
+ */
+#define MCHP_XEC_AF0		0x0
+#define MCHP_XEC_AF1		0x1
+#define MCHP_XEC_AF2		0x2
+#define MCHP_XEC_AF3		0x3
+#define MCHP_XEC_AF4		0x4
+#define MCHP_XEC_AF5		0x5
+#define MCHP_XEC_AF6		0x6
+#define MCHP_XEC_AF7		0x7
+
+#define MCHP_XEC_FUNC_MAX	(MCHP_XEC_AF7 + 1)
+
+/**
+ * @brief Macro to generate pinmux int using port, pin number and mode arguments
+ * This is taken from Linux equivalent st,stm32f429-pinctrl binding
+ * Microchip XEC documentation does not use letters for port naming instead
+ * using pin number ranges such as 000_036, etc. Use the port defines from
+ * mchp-xec-pinctrl-common: MCHP_XEC_PORT_000_036, etc.
+ * mode is AF0,..,AF7
+ * MCHP groups 32 pins per port requiring a 5-bit field for pin position.
+ */
+#define PIN_POS(pin)		((pin) % 32)
+#define PIN_NO(port, line)	((port) * 0x20 + (line))
+
+#define MCHP_XEC_PINMUX(port, pin, mode)				\
+	(((PIN_NO(MCHP_XEC_PORT_ ## port, PIN_POS(pin))) << 8) |	\
+	 (MCHP_XEC_ ## mode))
+
+/**
+ * @brief PIN configuration bitfield
+ *
+ * Pin configuration is coded with the following
+ * fields
+ *    Alternate Functions [ 0 : 3 ]
+ *    GPIO Input          [ 4 ]
+ *    GPIO Output         [ 5 ]
+ *    GPIO Output type    [ 6 ]
+ *    GPIO slew rate      [ 7 ]
+ *    GPIO Drive strength [ 8 : 9 ]
+ *    GPIO PUPD config    [ 10 : 11 ]
+ *
+ * Applicable to Microchip XEC series
+ */
+
+/* Alternate functions */
+#define MCHP_XEC_FUNC_ALT_0	0
```

Comment:
suggest an enum

Reply:
@jackrosenthal , Same DTC error as above if an enum is used.